__tests__/ord.test.js

```diff
@@ -13,4 +13,37 @@ describe("Tests for default ORD document", () => {
     const document = ord(csn);
     expect(document).toMatchSnapshot();
   });
+
+
+  describe("eventResources", () => {
+    const GROUP_ID_REGEX = /^([a-z0-9-]+(?:[.][a-z0-9-]+)*):([a-zA-Z0-9._\-/]+):([a-z0-9-]+(?:[.][a-z0-9-]+)*):(?<service>[a-zA-Z0-9._\-/]+)$/
+
+    let document;
+
+    beforeAll(()=> {
+      document = ord(csn);
+    })
+
+    test("Assigned to excactly one CDS Service group", () => {
+      for (const eventResource of document.eventResources) {
+        expect(eventResource.partOfGroups.length).toEqual(1)
+      }
+    });
+  
+    test("The CDS Service Group ID includes the CDS Service identifier", () => {
+      for (const eventResource of document.eventResources) {
+        const [groupId] = eventResource.partOfGroups
+        expect(groupId).toMatch(GROUP_ID_REGEX)
+
+        const match = GROUP_ID_REGEX.exec(groupId)
+        if (match && match.groups?.service) {
+          let service = match.groups?.service
+          if (service.startsWith("undefined")) service = service.replace("undefined.", "")
```

Comment:
Could you explain these two lines? And service seems to imply to serviceName?

Reply:
Yes, service imply serviceName, you can see it in the group id regex as a named capturing group.

The implementation here is not finished but my goal was to find the corresponding service in the CSN definition using the information I have in the ORD Document. In this case, I want to find that the eventResource `partOfGroups` is assigned to a `groupId` that is actually a service in the CSN.

Since they are attaching the `namespace` in the groupId, I want to remove the namespace part here to find the corresponding CSN definition. In this case, `undefined`.

I will improve this to check for the namespace and remove the namespace instead of `undefined`.

I hope this answered the question.